src/mem.rs

```diff
@@ -0,0 +1,60 @@
+// NOTE Copied verbatim from the rlibc crate
```

Comment:
> The focus of this crate is to provide all of the runtime functions

Sure, but couldn't that bb accomplished by this crate using `rlibc` as a dependency on the platforms that need it? Then using this crate will pull in rlibc and it looks the same as if the code were implemented here?